pkg/kubelet/status/status_manager.go

```diff
@@ -262,12 +262,40 @@ func (m *manager) UpdatePodFromAllocation(pod *v1.Pod) (*v1.Pod, bool) {
 	return updatePodFromAllocation(pod, allocs)
 }
 
-func updatePodFromAllocation(pod *v1.Pod, allocs state.PodResourceAllocation) (*v1.Pod, bool) {
+/* func updatePodFromAllocation(pod *v1.Pod, allocs state.PodResourceAllocation) (*v1.Pod, bool) {
 	allocated, found := allocs[string(pod.UID)]
 	if !found {
 		return pod, false
 	}
 
+	updated := false
+	updateContainerResources := func(c *v1.Container) {
+		if cAlloc, ok := allocated[c.Name]; ok {
+			if !apiequality.Semantic.DeepEqual(c.Resources, cAlloc) {
+				if !updated {
+					pod = pod.DeepCopy()
+					updated = true
+				}
+				c.Resources = *cAlloc.DeepCopy()
+			}
+		}
+	}
+
+	for i := range pod.Spec.Containers {
+		updateContainerResources(&pod.Spec.Containers[i])
+	}
+	for i := range pod.Spec.InitContainers {
+		updateContainerResources(&pod.Spec.InitContainers[i])
+	}
+	return pod, updated
+} */
+
+// TODO(vibansal): Refactor this function to something above commented code.
```

Comment:
The above method doesn't work? It looks good to me.

This needs to be resolved before this PR can merge.

Reply:
Refactored the code properly.